src/main/java/duke/Task.java

```diff
@@ -0,0 +1,63 @@
+package duke;
+
+public class Task {
+
+        protected String description;
+        protected boolean isDone;
+
+        public Task(String description) {
+            this.description = description;
+            this.isDone = false;
+        }
+        /**
+         * getter for status icon.
+         *
+         */
+        public String getStatusIcon() {
+            return (isDone ? "X" : " "); // mark done task with X
+        }
+        /**
+        * getter for task icon.
+        *
+        */
+        public String getTaskIcon() {
+            return "T";
+        }
+
+        /**
+        * returns string representation of task.
+        *
+        */
+        public String ToString(){
+            return "[" + getTaskIcon() + "] " + "[" + getStatusIcon() + "] "+ description;
+        }
+        /**
+        *  marks task as done.
+        *
+        */
+        public void markAsDone(){
+            isDone = true;
+        }
+        /**
+        *   marks task as undone.
+        *
+        */
+        public void unmarkAsDone(){
+        isDone = false;
```

Comment:
The preview on Github shows some strange indentation here? Seems to me that the last brace `}` is pushed too far to the left, and so is the line `isDone = false;`